src/Container/Lifetime/LifetimeContainer.cs

```diff
@@ -142,7 +173,10 @@ protected virtual void Dispose(bool disposing)
         /// </returns>
         public IEnumerator<object> GetEnumerator()
         {
-            return _items.GetEnumerator();
+            lock (_items)
```

Comment:
This is not related to allocations, just ReSharper said that in all other cases is used synchronized , only here without lock.